File: VB/WinSolution.Module/Order.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinSolution.Module
	<DefaultClassOptions> _
	Public Class Order
		Inherits BaseObject
		Implements ISimpleBusinessAction
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private nameCore As String
		Public Property Name() As String
			Get
				Return nameCore
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", nameCore, value)
			End Set
		End Property
		Private descriptionCore As String
		Public Property Description() As String
			Get
				Return descriptionCore
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Description", descriptionCore, value)
			End Set
		End Property
		<Persistent("Active")> _
		Private activeCore As Boolean = True
		<PersistentAlias("activeCore")> _
		Public ReadOnly Property Active() As Boolean Implements ISimpleBusinessAction.Active
			Get
				Return activeCore
			End Get
		End Property
		<Action> _
		Public Sub SimpleBusinessAction() Implements ISimpleBusinessAction.SimpleBusinessAction
			activeCore = Not activeCore
			OnChanged("Active")
		End Sub
	End Class
	Public Interface ISimpleBusinessAction
		ReadOnly Property Active() As Boolean
		Sub SimpleBusinessAction()
	End Interface
End Namespace

File: CS/WinSolution.Module/Order.cs
using System;
using System.ComponentModel;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinSolution.Module {
    [DefaultClassOptions]
    public class Order : BaseObject {
        public Order(Session session) : base(session) { }
        private string nameCore;
        public string Name {
            get { return nameCore; }
            set { SetPropertyValue("Name", ref nameCore, value); }
        }
        private string descriptionCore;
        public string Description {
            get { return descriptionCore; }
            set { SetPropertyValue("Description", ref descriptionCore, value); }
        }
        [Persistent("Active")]
        private bool activeCore = true;
        [PersistentAlias("activeCore")]
        public bool Active {
            get { return activeCore; }
        }
        [Action]
        public void SimpleBusinessAction() {
            activeCore = !activeCore;
            OnChanged("Active");
        }
    }
}